linera-core/src/unit_tests/client_test_utils.rs

```diff
@@ -721,6 +737,104 @@ impl StorageBuilder for MakeRocksDbStorage {
     }
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+pub struct MakeServiceStorage {
+    _guard: Option<StorageServiceGuard>,
+    endpoint: String,
+    namespace: String,
+    use_child: bool,
+    instance_counter: usize,
+    wasm_runtime: Option<WasmRuntime>,
+    clock: TestClock,
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl Default for MakeServiceStorage {
+    fn default() -> MakeServiceStorage {
+        let _guard = None;
+        let clock = TestClock::default();
+        let endpoint = "127.0.0.1:8742".to_string();
+        let namespace = generate_test_namespace();
+        let use_child = true;
+        Self {
+            _guard,
+            endpoint,
+            namespace,
+            use_child,
+            instance_counter: 0,
+            wasm_runtime: None,
+            clock,
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl MakeServiceStorage {
+    /// Creates a `ServiceStorage` from just an endpoint
+    pub fn new(endpoint: &str) -> Self {
+        let _guard = None;
+        let endpoint = endpoint.to_string();
+        let clock = TestClock::default();
+        let namespace = generate_test_namespace();
+        let use_child = true;
+        Self {
+            _guard,
+            endpoint,
+            namespace,
+            use_child,
+            instance_counter: 0,
+            wasm_runtime: None,
+            clock,
+        }
+    }
+
+    /// Creates a `ServiceStorage` from an endpoint and the wasm runtime.
+    pub fn with_wasm_runtime(endpoint: &str, wasm_runtime: impl Into<Option<WasmRuntime>>) -> Self {
+        let _guard = None;
+        let endpoint = endpoint.to_string();
+        let clock = TestClock::default();
+        let namespace = generate_test_namespace();
+        let use_child = true;
+        Self {
+            _guard,
+            endpoint,
+            namespace,
+            use_child,
+            instance_counter: 0,
+            wasm_runtime: wasm_runtime.into(),
+            clock,
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+#[async_trait]
+impl StorageBuilder for MakeServiceStorage {
+    type Storage = ServiceStorage<TestClock>;
+
+    async fn build(&mut self) -> Result<Self::Storage, anyhow::Error> {
+        if self._guard.is_none() && self.use_child {
+            let binary = get_service_storage_binary().await?.display().to_string();
+            let spanner = StorageServiceSpanner::new(&self.endpoint, binary);
```

Comment:
(I'm still not sure I understand why it's called a "spanner".)

Reply:
I renamed it `StorageServiceBuilder`.